start

```diff
@@ -199,8 +206,12 @@ function copy_defaults() {
     echo "supervisor: config directory exists, skipping copy"
   else
     $CP /opt/stellar-default/common/supervisor/ $SUPHOME
-    if [ -d /opt/stellar-default/$NETWORK/supervisor/ ]; then
-      $CP /opt/stellar-default/$NETWORK/supervisor/ $SUPHOME
+    echo "copied default stuff"
+    if [ "$ENABLE_SOROBAN_RPC" = "true" ] && [ -f /opt/stellar-default/$NETWORK/supervisor/etc/supervisord.conf.d/soroban-rpc.conf ]; then
+      cp /opt/stellar-default/$NETWORK/supervisor/etc/supervisord.conf.d/soroban-rpc.conf $SUPHOME/etc/supervisord.conf.d
+    fi
+    if [ -f /opt/stellar-default/$NETWORK/supervisor/etc/supervisord.conf.d/friendbot.conf ]; then
+      cp /opt/stellar-default/$NETWORK/supervisor/etc/supervisord.conf.d/friendbot.conf $SUPHOME/etc/supervisord.conf.d
```

Comment:
These lines look unnecessary as friendbot should only be usable on standalone.

Reply:
I cleaned up the duplication aspect, I can't let a blanket recursive copy of network folder($NETWORK/supervisor/etc/supervisord.conf.d) be done, because the runtime flag check on $ENABLE_SOROBAN_RPC makes the choice of whether the supervisor cfg should be copied to be more dynamic, I refactored it slightly to process each optional service via it's conf file agnostic of network type. this should allow testnet/pubnet to copy soroban-rpc.conf's at some point later when those network folders are updated to have the .conf to launch it.